### ISSUE TITLE
Read Me clarification

### ISSUE THREAD

**Issue by moda20** [krista-koivisto/isolate_handler#8]
I have read the following in your README :

> It is worth noting that running native plugins from a Dart isolate does not offer any real performance advantage as all native code is run on the main (UI) thread by default. Your UI and main thread can be much more responsive if you use it properly but you cannot expect the platform channel calls to run in parallel.

What does native plugins mean ? u mean the ones that are made by flutter team originally ?

what does "as all native code is run on the main (UI) thread by default" mean ? did you mean that all Native plugin code is run Outside the main UI thread by default ? 

"you cannot expect the platform channel calls to run in parallel."  did you mean that you win't be able to parallelize the platform channels themselves (on or outside isolates) or running them on parallel with the main UI thread ?
If you can, please explain that paragraph a bit more.

**Comment by raccoons-are-goofy** [comment]
I was the previous owner of this package so perhaps I can shed some light on the situation.

*Native* refers to the native platform, typically iOS or Android. *Native plugin* refers to any plugin which executes custom code on one or more native platforms. By default, this native code is run on the same thread as the UI.

There is a subtle, but hugely important distinction to make here: native platform threads are separate from isolates. Creating a new isolate will not create a new native thread. Therefore, whenever you execute plugin code, it is done so on the platform's main thread (the thread the UI runs on) whether you do so from an isolate or not.

In order to avoid compute-intensive native code freezing up the UI, a thread must be created on the native platform instead.

**TL;DR** Creating a new isolate will not create a new native thread. If you have a plugin that is compute intensive on the native side, running it in a Dart/Flutter isolate will not improve its performance. A native thread is required for that.

Does that clear things up a bit?

**Comment by moda20** [comment]
yeah it did, so from what I understood the dart isolates only parallelize the dart code but still call the same native thread when using the MethodChannels to communicate. 

a quick question : A newly created native thread would be a whole bare bone flutter application or just a native thread to execute native code in parallel ?

**Comment by raccoons-are-goofy** [comment]
Yes, that's correct.

As for your quick question, a new thread is just a new native thread to execute code in parallel. You just create a new thread the way you would if you were developing for that platform without Flutter.

**Comment by deakjahn** [comment]
I changed the README to add these thoughts and lifted the discussion up from the end.
